lib/ansible/inventory/group.py

```diff
@@ -114,6 +114,11 @@ def add_host(self, host):
         host.add_group(self)
         self.clear_hosts_cache()
 
+    def remove_host(self, host):
+        host.remove_group(self)
+        self.hosts.remove(host)
+        self.clear_hosts_cache()
```

Comment:
At first I thought this might be overkill, but all hosts need to rebuild their groups vars to remove this host from them. Making note in case someone else goes down same road.